kornia/augmentation/base.py

```diff
@@ -308,9 +327,10 @@ def transform_boxes(
             self._params if params is None else params, flags, **kwargs
         )
 
-        to_apply = params['batch_prob']
+        batch_prob = params['batch_prob']
+        to_apply = batch_prob > 0.5  # NOTE: in case of Relaxed Distributions.
         output: Boxes
-        if to_apply.all():
+        if to_apply.bool().all():
```

Comment:
this already is a bool tensor, i don't think this needs to cast
```suggestion
        if to_apply.all():
```